main.cpp

```diff
@@ -1,25 +1,284 @@
+string prev_file;   //nazwa poprzedniej mapy
+#define SIZE 100
+string mapa[SIZE], bufor[SIZE];
+char user_ch='X', end_ch='O';
+COORDS player_coords;
+COORDS end_coords;
 
 // procedury
-void getMap(); // wczytuje mapę z pliku file_name i zapisuje do mapa[]
-void beginingcoords(); // bierze poczontkowew koordy i wrzuca do zmiennej
-void targetcoords(); // bierze koordy celu/wyjścia i wrzuca do zmiennej
-bool isWall(int x, int y);  // czy na podanych kordach nie ma spacji
-char drawOnBufor(int x, int y, char c); // narysuj na x i y znak c i zwróć poprzednie co tam było
-void viewBufor(); // wypisz bufor na ekran
-char getEvent(); // pobierz znak
-void doEvent(); // wykonaj operację przypisaną do danego znaku (np WSAD)
+void getMap()
+{
+std::fstream file;
+file.open(file_name, ios::in);
+
+if(file.good()==false)
+{
+    throw "File does not exist";
+   
+}
+string line;
+size_t i=0;
+while (getline(file, line))
+{        
+    mapa[i] = line;
+    i++;
+}
+
+file.close();
+}
+
+bool isWall(COORDS c)  // czy na podanych kordach nie ma spacji
+{
+    if (mapa[c.y][c.x] == ' ') //sprawdza czy space
+        return false;                //zwraca, czy jest, czy nie
+        else return true;
+}
+
+bool isExist(COORDS p) { // sprawdza czy dane pole jest na mapie
+  if(p.y>=SIZE)return false;
+  return mapa[p.y].size()>=p.x;
+}
+COORDS randomCoords(bool notWall=true) {
+  COORDS c;
+  do {
+    c.y = rand() % SIZE;
+    size_t s = mapa[c.y].size();
+    if(s) // s != 0
+      c.x = rand() % s;
+  }while(!(isExist(c)&&((!notWall)||isWall(c))));
+  return c;
+}
+void beginingcoords() { // bierze poczontkowe koordy i wrzuca do zmiennej
+  player_coords = randomCoords();
+}
+void targetcoords() { // bierze koordy celu/wyjścia i wrzuca do zmiennej
+  end_coords = randomCoords();
+}
+char getEvent()
+    {
+        return getch();
+    }// pobierz znak
+void doEvent(char c) {
+    c = toupper(c);
+    COORDS n=player_coords;
+    switch(c) {
+        case 'W': ++n.y; break;
+        case 'S': --n.y; break;
+        case 'A': --n.x; break;
+        case 'D': ++n.x; break;
+    }
+    if(!isWall(n))
+        player_coords = n;
+
+}// wykonaj operację przypisaną do danego znaku (np WSAD)
+char drawOnBufor(COORDS C, char c) { // narysuj na x i y znak c i zwróć poprzednie co tam było
+  char oc = mapa[C.y][C.x];
+  mapa[C.y][C.x] = c;
+  return oc;
+}
+void refreshBufor() { // załaduj mapę do bufora i nanieś na nie usera i wyjście
+  for(size_t i=0; i<SIZE; ++i)
+    bufor[i] = mapa[i];
+  drawOnBufor(player_coords, user_ch);
+  drawOnBufor(end_coords, end_ch);
+}
+void viewBufor() { // wypisz bufor na ekran
+  if(system(NULL)) // sprawdzanie czy konsola dostępna
+    if(system("CLS")) // sprawdzanie czy komenda CLS zadziałała
+      if(system("clear"))
+        cout << string('\n',0xFF);
+  for(size_t i=0;i<SIZE;++i)
+    if(mapa[i].size()>0)
+      cout << mapa[i] << '\n';
+}
 bool isEnd(); // czy jesteśmy na kordach wyjścia
```

Comment:
/tmp/ccvkiu8A.o: In function `main':
main.cpp:(.text+0xdd5): undefined reference to `isEnd()'
collect2: error: ld returned 1 exit status
GET 1

Reply:
@Crackhoff jak tej funkcji nie było... tak nadal nie ma

Reply:
już jest